### ISSUE TITLE
[FR.] Optional kafka message tagging and tag filtering

### ISSUE THREAD

**Issue by yjqg6666** [pinpoint-apm/pinpoint#8263]
**Which module is your feature request related to?**
Plugin

**Is your feature request related to a problem?**
Use pinpoint kafka plugin to add a message tag header from producer and filter messages by tag header on consumer side.

**Describe the solution you'd like**
On the producer side, read tag config, add tag header to message when producing messages if configured. 
On the consumer side, read tag filtering config, filtering consumer messages when constructing the ConsumerRecords if configured.

A PR would be made for this feature.

**Comment by koo-taejin** [comment]
@yjqg6666 

I have a question.
I looked at PR, as faf as I have understood, it seems to be filtering the actual message, not the trace, is my understanding correct?

**Comment by yjqg6666** [comment]
@koo-taejin You are right. It's tagging messages on producer sides and filtering messages based on tags on them and tag filtering configuration on the consumer sides.

**Comment by yjqg6666** [comment]
@koo-taejin Understood.

**Comment by koo-taejin** [comment]
@yjqg6666 

I had discussed about this PR. we made concultion that Pinpoint should not change method processing of framework or library.
For above reason, we could not accept this PR. 
But If you share what you want to do, I am going to think the way we solve that problem. 

I sincerely ask for your understanding.

Thanks.
